bindings/pyroot/cppyy/CPyCppyy/src/Converters.cxx

```diff
@@ -1464,50 +1481,66 @@ bool CPyCppyy::VoidArrayConverter::ToMemory(PyObject* value, void* address)
     return true;
 }
 
+
+//----------------------------------------------------------------------------
+static inline void init_shape(Py_ssize_t defdim, dims_t dims, Py_ssize_t*& shape)
+{
+    int nalloc = (dims && 0 < dims[0]) ? (int)dims[0]+1: defdim+1;
+    shape = new Py_ssize_t[nalloc];
+    if (dims) {
+        for (int i = 0; i < nalloc; ++i)
+            shape[i] = (Py_ssize_t)dims[i];
+    } else {
+        shape[0] = defdim;
+        for (int i = 1; i < nalloc; ++i) shape[i] = UNKNOWN_SIZE;
+    }
+}
+
 //----------------------------------------------------------------------------
 #define CPPYY_IMPL_ARRAY_CONVERTER(name, ctype, type, code)                  \
-CPyCppyy::name##ArrayConverter::name##ArrayConverter(dims_t dims) {          \
-    int nalloc = (dims && 0 < dims[0]) ? (int)dims[0]+1: 2;                  \
-    fShape = new Py_ssize_t[nalloc];                                         \
-    if (dims) {                                                              \
-        for (int i = 0; i < nalloc; ++i) fShape[i] = (Py_ssize_t)dims[i];    \
-    } else {                                                                 \
-        fShape[0] = 1; fShape[1] = -1;                                       \
-    }                                                                        \
+CPyCppyy::name##ArrayConverter::name##ArrayConverter(dims_t dims, bool init) {\
+    if (init) {                                                              \
+        init_shape(1, dims, fShape);                                         \
+        fIsFixed = fShape[1] != UNKNOWN_SIZE;                                \
+    } else fIsFixed = false;                                                 \
 }                                                                            \
                                                                              \
 bool CPyCppyy::name##ArrayConverter::SetArg(                                 \
-    PyObject* pyobject, Parameter& para, CallContext* /* ctxt */)            \
+    PyObject* pyobject, Parameter& para, CallContext* ctxt)                  \
```

Comment:
Extra whitespaces

Reply:
Needed because of the `\` :smiley:

Reply:
Ah, sorry! The highlighting on github tricked me :sweat_smile: